core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -393,7 +393,7 @@ object SparkEnv extends Logging {
     // Add a reference to tmp dir created by driver, we will delete this tmp dir when stop() is
     // called, and we only need to do it for driver. Because driver may run as a service, and if we
     // don't delete this tmp dir when sc is stopped, then will create too many tmp dirs.
-    if (isDriver) {
+    if (isDriver && conf.getOption("spark.submit.deployMode").getOrElse("client") == "client") {
```

Comment:
Tiny nitpick, this might be simpler.

```
  if (isDriver && conf.get("spark.submit.deployMode", "client") == "client") { 
```

Reply:
Ok, thanks, I will change it.

Reply:
Originally, my version is
```
conf.get("spark.submit.deployMode", "client") == "client"
```
Then I refered to the SparkContext#deployMode function,
it use
```
conf.getOption("spark.submit.deployMode").getOrElse("client")
```
I just want to keep the same style as SparkContext.
which one is more better?